npm/src/controller/connection/saml.ts

```diff
@@ -17,6 +17,18 @@ import {
 import saml20 from '@boxyhq/saml20';
 import x509 from '../../saml/x509';
 import { JacksonError } from '../error';
+import axios from 'axios';
+
+async function fetchMetadata(resource: string) {
+  const response = await axios(resource, {
+    maxContentLength: 1000000,
+    maxBodyLength: 1000000,
+    timeout: 8000,
+  }).catch(() => {
+    throw new JacksonError("Couldn't fetch XML data", 400);
```

Comment:
It would be better to relay the status code of the axios fetch (`error.response.status`), instead of 400.